cmd/controller-manager/controller-manager.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	bucketcontroller "github.com/kubernetes-sigs/container-object-storage-interface-api/controller"
+	"github.com/kubernetes-sigs/container-object-storage-interface-controller/pkg/bucketrequest"
+
+	"github.com/golang/glog"
+)
+
+var cmd = &cobra.Command{
+	Use:           "controller-manager",
+	Short:         "central controller for managing bucket* and bucketAccess* API objects",
+	SilenceErrors: true,
+	SilenceUsage:  true,
+	RunE: func(c *cobra.Command, args []string) error {
+		return run(c.Context(), args)
+	},
+	DisableFlagsInUseLine: true,
+}
+
+var kubeConfig string
+
+func init() {
+	viper.AutomaticEnv()
+
+	cmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
+	flag.Set("logtostderr", "true")
+
+	strFlag := func(c *cobra.Command, ptr *string, name string, short string, dfault string, desc string) {
+		c.PersistentFlags().
+			StringVarP(ptr, name, short, dfault, desc)
+	}
+	strFlag(cmd, &kubeConfig, "kube-config", "", kubeConfig, "path to kubeconfig file")
+
+	hideFlag := func(name string) {
+		cmd.PersistentFlags().MarkHidden(name)
+	}
+	hideFlag("alsologtostderr")
+	hideFlag("log_backtrace_at")
+	hideFlag("log_dir")
+	hideFlag("logtostderr")
+	hideFlag("master")
+	hideFlag("stderrthreshold")
+	hideFlag("vmodule")
+
+	// suppress the incorrect prefix in glog output
+	flag.CommandLine.Parse([]string{})
+	viper.BindPFlags(cmd.PersistentFlags())
+
+}
+
+func main() {
+	if err := cmd.Execute(); err != nil {
+		glog.Fatal(err.Error())
+	}
+
+	var cancel context.CancelFunc
+
+	_, cancel = context.WithCancel(cmd.Context())
```

Comment:
`cmd` would not 'have' a `Context` here since we never set it. beforehand.

Reply:
By default cmd gets context.Background() right?

Reply:
Yes, though that's a bit of a 'fallback' in `cobra`. I think being explicit here (also see snippet above) is preferable.